src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -141,6 +141,36 @@ internal bool IsExceptionType(TypeSymbol type, ref HashSet<DiagnosticInfo> useSi
             return IsEqualOrDerivedFromWellKnownClass(type, WellKnownType.System_Exception, ref useSiteDiagnostics);
         }
 
+        internal bool IsWellKnownTupleType(TypeSymbol typeToCheck, int requiredArity)
+        {
+            // NOTE: error type symbol is NamedTypeSymbol, 
+            //       but not SymbolKind.NamedType, so check the kind before casting.
+            if (typeToCheck.Kind != SymbolKind.NamedType)
+            {
+                return false;
+            }
+
+            NamedTypeSymbol currentType = (NamedTypeSymbol)typeToCheck;
+            const int maxArity = TupleTypeSymbol.RestPosition;
```

Comment:
added an assert. The helper is meant to be used to validate underlying types, not with actual tuples.